lms/routes.py

```diff
@@ -29,6 +29,15 @@ def includeme(config):  # pylint:disable=too-many-statements
 
     config.add_route("api.assignments.create", "/api/assignment", request_method="POST")
 
+    # The gateway end-point is a bit of an oddball as it uses LTI launch as the
+    # authentication method, and so requires the LTILaunch resource
```

Comment:
Does it require the context authentication of convince?

I reckon the security part will be `LTISecurityPolicy` jobs and having the resource here will make `contex.application_instance` and friend available.

The security stuff it's a bit convoluted so I might be getting it wrong.

Reply:
I _think_ the context stuff happens after authentication. This makes `context.application_instance` available to us easily in the view.